Source/WebCore/html/HTMLCanvasElement.cpp

```diff
@@ -584,8 +582,8 @@ void HTMLCanvasElement::reset()
     int w = limitToOnlyHTMLNonNegative(attributeWithoutSynchronization(widthAttr), defaultWidth);
     int h = limitToOnlyHTMLNonNegative(attributeWithoutSynchronization(heightAttr), defaultHeight);
 
-    if (is<CanvasRenderingContext2D>(m_context))
-        downcast<CanvasRenderingContext2D>(*m_context).reset();
+    if (RefPtr context = dynamicDowncast<CanvasRenderingContext2D>(m_context.get()))
```

Comment:
do we need to take a ref here?

this replacement pattern is repeated multiple times, maybe worth a mention in the commit log as it's not just a replacement with dynamicDowncast

Reply:
I will mention in the commit log. I did a little of smart pointer adoption in the patch too, on the lines I changed. Here we're calling a non-trivial function on the context so we're expected to be holding a local RefPtr/Ref to the context.